src/App.ts

```diff
@@ -304,6 +304,9 @@ export default class App {
 
     // Check for required arguments of HTTPReceiver
     if (receiver !== undefined) {
+      if (this.socketMode) {
+        throw new AppInitializationError('You cannot pass receiver when socketMode is set to true');
```

Comment:
Any suggestions on this error message?

Reply:
Per the issue:

> currently the App constructor **quietly ignores** the `receiver: Receiver` argument when `socketMode: boolean` as true is given

If this is true, won't introducing an exception cause issues for folks that have been otherwise been able to operate with both arguments passed? If so, perhaps a warning would suffice, at least giving them notice that this combination of options results in `receiver` being ignored.

Reply:
Thanks for the suggestion! 

I am still thinking that we can throw an exception here. There are two reasons:
1. The `App` constructor validation consistency (We have no warnings over the exception in the `App` constructor so far)
1. The severity of ignorance of the explicitly passed `receiver` (If the `receiver` has custom logic parts, knowing this misconfiguration as early as possible is crucial)

With that being said, if other maintainers think that we can go with warning logs first and then switch to initialization error, I'm happy to change the validation logic in the way.

Reply:
I feel like throwing an exception is an acceptable approach here. It may require that _some_ developers change their code, but in the name of clarity it seems a reasonable and simple change.

Reply:
Exception it is! As for a message suggestion: `receiver cannot be passed when socketMode is set to true`

Reply:
@misscoded Thanks for the suggestion:
```suggestion
        throw new AppInitializationError('receiver cannot be passed when socketMode is set to true');
```